File: NuzgulPalindromeConfuser (Python)/nuzgulConfuser.py
#!/usr/bin/python
import sys

# recursive palindrome finder function
def isPalindrome(palin):
    if len(palin) == 0 or len(palin) == 1:
        return True
    else:
        if palin[0] != palin[len(palin) - 1]:
            return False
        else:
            return isPalindrome(palin[1:-1])

# main
def main():

	# ensures argument is passed
	try:
	    fileName = sys.argv[1]
	except IndexError:
		print "File argument not given."
		sys.exit()

	# ensures file opens
	try:
		file = open(fileName, 'r')
	except IOError:
		print "Unable to open file! Check if file exists."
		sys.exit()

	for line in file:
		# removes whitespace
		line = "".join(line.split())
		# discards empty lines	
		if len(line) != 0:				
			if isPalindrome(line):
				print "AY",
			else:
				print "NAY",
			# converts to list lexically sorts in reverse order and converts back to string
			print "| " + "".join(sorted(list(line), reverse=True)) 
	file.close()

main()

# TODO unit tests

File: NuzgulPalindromeConfuser (Rust)/NuzgulConfuser.rs
use std::io::BufReader;
use std::fs::File;
use std::io::prelude::*;
use std::iter::Iterator;
use std::iter::FromIterator;
use std::env;

// recursive palindrome finder function
fn is_palindrome(palin: String) -> bool {
	let size = palin.len();
	if size == 0 || size == 1 {
        return true;
    }
    else {
        if palin.chars().nth(0).unwrap() != palin.chars().nth(size - 1).unwrap() {
            return false;
            }
        else {
        	let p = &palin[1..size-1];
            return is_palindrome(p.to_string());
    	}
    }
}

//main
fn main() {
	let args: Vec<_> = env::args().collect();
    if args.len() > 1 {
        let ref file_name = args[1];
        let file = BufReader::new(File::open(file_name).expect("Unable to open file! Check if file exists."));
	    for line in file.lines().filter_map(|result| result.ok()) {
	    	if line != "" {
		    	let ref palin1 = line.replace(" ", "");
		    	let palin2 = palin1.replace("	", "");
		    	let palin3 = palin2.replace("", "");
		    	if is_palindrome(palin2) == true {
		    		print!("AY")
		    	}
		    	else {
		    		print!("NAY")
		    	}
		    	// lexical reverse sorting
			    let s_slice: &str = &palin3[..];
			    let mut chars: Vec<char> = s_slice.chars().collect();
			    chars.sort_by(|a, b| b.cmp(a));
			    let s = String::from_iter(chars);
			    println!(" | {}", s);
	    	}
	    }
    }
    // ensures argument is passed
    else {
    	print!("File argument not given.");
    	 std::process::exit(0);
    }
}